src/NServiceBus.Core/Routing/AssemblyRouteSource.cs

```diff
@@ -20,7 +20,7 @@ public AssemblyRouteSource(Assembly messageAssembly, UnicastRoute route)
         public IEnumerable<RouteTableEntry> GenerateRoutes(Conventions conventions)
         {
             var routes = messageAssembly.GetTypes()
-                .Where(t => conventions.IsMessageType(t))
+                .Where(conventions.IsMessageType)
```

Comment:
I don't see how further applications of this would have any impact of the "correctness"? Also, I'm not saying this is wrong, I think brandon is right by stating this version is less efficient and it seems to be a trivial thing to change it to be more efficient (and I admit that I prefer your version better from a readability perspective, but so do I prefer LINQ over foreach...)?